cfg/builder/builder_walk.cc

```diff
@@ -169,6 +169,14 @@ BasicBlock *CFGBuilder::walk(CFGContext cctx, ast::Expression *what, BasicBlock
                  */
             },
             [&](ast::Return *a) {
+                if (!ast::isa_tree<ast::EmptyTree>(a->expr.get()) &&
+                    core::Types::isSubType(cctx.ctx, cctx.inWhat.symbol.data(cctx.ctx)->resultType,
+                                           core::Types::void_())) {
+                    if (auto e = cctx.ctx.state.beginError(a->loc, core::errors::CFG::ReturnExprVoid)) {
+                        e.setHeader("`{}` returns `void` but contains an explicit `return <expression>`",
```

Comment:
Some suggestions:

- rewriting this as like this:

```
"... explicit `{}` ...", "return"
```

will make it change colors instead of show the backticks when colors are available.

- can we make it explicit like "explicitly returns an expression" or something? I'd not like to have `<expression>` in our error message